skymp5-server/cpp/server_guest_lib/PieScript.cpp

```diff
@@ -121,13 +256,42 @@ uint32_t PieScript::GetSlotItem(int weaponChance, int armoryChacne,
     weaponChance, armoryChacne, consumableChance, nothingChance);
   int item = GenerateRandomNumber(
     0, lootTable[typeAndTier.first][typeAndTier.second].size() - 1);
-  return lootTable[typeAndTier.first][typeAndTier.second][item];
+  return lootTable[typeAndTier.first][typeAndTier.second].at(item);
```

Comment:
`lootTable[typeAndTier.first][typeAndTier.second]` might crash the server if it's array/vector, please throw an exception if it's out of range, or use `.at` too